[17.0][MIG] product_assortment: Migration to 17.0

dalonsod opened the pull request:
~~Backport from v18 #1904~~ Standard migration from v16 + additional basic security management added from v18 implementation.

There are three unfinished and closed PRs for this migration (#1540 , #1587  and #1613), I hope this is the last one...

rousseldenis committed:
[10.0][ADD] product_assortment

tbaden committed:
[FIX][10.0] product_assortment travis warning

rousseldenis committed:
[12.0][MIG] product_assortment

Commit:
[IMP] product_assortment: black, isort

Commit:
[MIG] product_assortment: Migration to 13.0

benwillig committed:
[FIX] avoid errors when trying to open assortment products

sbidoul committed:
[FIX] product_assortment: development_status

CarlosRoca13 committed:
[IMP] product_assortment: added partners to products assortments

pedrobaeza committed:
[FIX] product_assortment: Avoid errors on invalid models

You can have an ir.filters entry with an invalid model. That shouldn't
break system in case of accessing the filter.

claudiagn committed:
Added translation using Weblate (Catalan)

claudiagn committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (20 of 20 strings)

Translation: product-attribute-13.0/product-attribute-13.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-13-0/product-attribute-13-0-product_assortment/ca/

xavier-bouquiaux committed:
[14.0][MIG] product_assortment

rousseldenis committed:
[14.0][IMP] product_assortment: Fallback to create() for model_id

Kev-Roche committed:
fix ir_filters

sergio-teruel committed:
[IMP] product_assortment: Allow to use assortment based on partner domain
TT30584

CarlosRoca13 committed:
[IMP] product_assortment: add icon to menu

Commit:
[FIX] remove useless depends on domain leading to orm crash

hailangvn committed:
[FIX] product_assortment: use safe_eval.datetime

sebalix committed:
[FIX] Fix access right issues when using actions

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/ca/

SicurSam committed:
Added translation using Weblate (Italian)

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 70.8% (17 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/it/

SicurSam committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (24 of 24 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_assortment/it/

CarlosRoca13 committed:
[MIG] product_assortment: Migration to 15.0

sergio-teruel committed:
[IMP] product_assortment: Add blacklist product domain to exclude some products
TT36814

marielejeune committed:
[MIG] product_assortment: Migration to 16.0

lmignon committed:
[MIG] product_assortment: Fix form view

Commit:
[UPD] Update product_assortment.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
product_assortment 16.0.1.0.1

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/

marylla committed:
Added translation using Weblate (German)

marylla committed:
Translated using Weblate (German)

Currently translated at 92.5% (25 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/de/

marielejeune committed:
[FIX] product_assortment unit tests: invalidate_cache() is deprecated

OCA-git-bot committed:
product_assortment 16.0.1.0.2

Commit:
[UPD] Update product_assortment.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 66.6% (18 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

carlosdauden committed:
[FIX] product_assortment: Product black_list_domain invalidates whitelist_product_ids

carlosdauden committed:
[IMP] product_assortment: Add new methods to improve performance using ormcache

PicchiSeba committed:
Translated using Weblate (Italian)

Currently translated at 74.0% (20 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

PicchiSeba committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

carlosdauden committed:
[IMP] product_assortment: New field applied_assortments_ids to avoid cache "inefficiency" using workers because cache is isolated by worker

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/es/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (27 of 27 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

Commit:
[UPD] Update product_assortment.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (29 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/it/

samibc2c committed:
Added translation using Weblate (French)

samibc2c committed:
Translated using Weblate (French)

Currently translated at 93.1% (27 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/fr/

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 93.1% (27 of 29 strings)

Translation: product-attribute-16.0/product-attribute-16.0-product_assortment
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-16-0/product-attribute-16-0-product_assortment/nl/

santostelmo committed:
[FIX] product_assortment ir_filters create vals_list

OCA-git-bot committed:
[BOT] post-merge updates

dalonsod commented:
@Deriman-Alonso @JasminSForgeFlow could you review? Thanks!

pedrobaeza commented:
There's a change in ORM in 18 that makes the module to behave different and there have been changes around, so I think it's better to migrate 16 version than to backport 18.

dalonsod committed:
[IMP] product_assortment: pre-commit auto fixes

dalonsod committed:
[MIG] product_assortment: Migration to 17.0

PieterPaulussen committed:
[IMP] product_assortment: basic management security support

Makes product assortment management access only available for a new
Product Assortment Manager security group

Co-Authored-By: David Alonso <david.alonso@solvos.es>

rlimaeco committed:
[IMP] product_assortment: Add test for assortment update

Co-Authored-By: David Alonso <david.alonso@solvos.es>

dalonsod commented:
> There's a change in ORM in 18 that makes the module to behave different and there have been changes around, so I think it's better to migrate 16 version than to backport 18.

@pedrobaeza done!

pedrobaeza commented:
/ocabot migration product_assortment

dalonsod commented:
@santostelmo @carolinafernandez-tecnativa could you review? Thanks!

rousseldenis reviewed (approved):
LGTM

Jaimermaccione reviewed (approved):
LGTM! Functional review on runboat.

HaraldPanten reviewed (approved):
/ocabot merge nobump
